From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct sctp_association
Patch-mainline: never, kabi
References: kabi

In networking-stable-18_12_12, upstream commit
fb6df5a6234c38a9c551559506a49a677ac6f07a (sctp: kfree_rcu asoc) added
rcu_head to struct sctp_association. It made the kABI checker to
complain.

Given the structure is allocated in sctp_association_new and the new
member as added to the end, hide the change from the kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/sctp/structs.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/include/net/sctp/structs.h
+++ b/include/net/sctp/structs.h
@@ -1899,8 +1899,9 @@ struct sctp_association {
 
 	__u64 abandoned_unsent[SCTP_PR_INDEX(MAX) + 1];
 	__u64 abandoned_sent[SCTP_PR_INDEX(MAX) + 1];
-
+#ifndef __GENKSYMS__
 	struct rcu_head rcu;
+#endif
 };
 
 
